src/odemis/gui/cont/tabs.py

```diff
@@ -3343,9 +3346,17 @@ def __init__(self, name, button, panel, main_frame, main_data):
                                        )
                 speccnt_spe = self._stream_controller.addStream(speccnts,
                                     add_to_view=self.panel.vp_align_fiber.view)
+                if main_data.tc_od_filter:
+                    speccnt_spe.add_axis_entry("density", main_data.tc_od_filter)
                 speccnt_spe.stream_panel.flatten()
                 self._speccnt_stream = speccnts
                 speccnts.should_update.subscribe(self._on_ccd_stream_play)
+
+                if fbdet2 is not None:
+                    self._fbdet2_txt = wx.TextCtrl(speccnt_spe.stream_panel, style=wx.BORDER_NONE | wx.TE_READONLY)
+                    speccnt_spe.add_txt_entry(speccnt_spe.stream_panel, "Detector 2", self._fbdet2_txt, 12)
+                    fbdet2.data.subscribe(self._on_tc_stream)
```

Comment:
You need to only subscribe to that detector when the fiber align mode is set, and unsubscribe when it's either not set or the tab is not shown.